src/Microsoft.ML.Data/Transforms/ValueMappingTransform.cs

```diff
@@ -0,0 +1,630 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Transforms;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+[assembly: LoadableClass(ValueMappingTransform.Summary, typeof(IDataTransform), typeof(ValueMappingTransform),
+    typeof(ValueMappingTransform.Arguments), typeof(SignatureDataTransform),
+    ValueMappingTransform.UserName, "ValueMapping", "ValueMappingTransform", ValueMappingTransform.ShortName,
+    DocName = "transform/ValueMappingTransform.md")]
+
+[assembly: LoadableClass(ValueMappingTransform.Summary, typeof(IDataTransform), typeof(ValueMappingTransform), null, typeof(SignatureLoadDataTransform),
+    "Value Mapping Transform", ValueMappingTransform.LoaderSignature)]
+
+[assembly: LoadableClass(ValueMappingTransform.Summary, typeof(ValueMappingTransform), null, typeof(SignatureLoadModel),
+    "Value Mapping Transform", ValueMappingTransform.LoaderSignature)]
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// The ValueMappingEstimator is a 1-1 mapping from a key to value. The key type and value type are specified
+    /// through TKeyType and TValueType. Arrays are supported for vector types which can be used as either a key or a value
+    /// or both. The mapping is specified, not trained by providiing a list of keys and a list of values.
+    /// </summary>
+    /// <typeparam name="TKeyType">Specifies the key type</typeparam>
+    /// <typeparam name="TValueType">Specifies the value type</typeparam>
+    public sealed class ValueMappingEstimator<TKeyType, TValueType> : TrivialEstimator<ValueMappingTransform<TKeyType, TValueType>>
```

Comment:
>ValueMappingEstimator [](start = 24, length = 21)

Would be nice to add example to our [https://github.com/dotnet/machinelearning/tree/master/docs/samples/Microsoft.ML.Samples/Dynamic](https://github.com/dotnet/machinelearning/tree/master/docs/samples/Microsoft.ML.Samples/Dynamic) samples project